### PR TITLE
Bug/e UI 6963

### PR DESCRIPTION
**Pull request by MateiHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6963

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/roleAccess/dtos/to-role-assignment-dto.ts [hunk 1]
```diff
@@ -87,7 +92,9 @@ export function toSARoleAssignmentBody(currentUserId: string, specificAccessData
   };
 }
 
-export function toDenySARoleAssignmentBody(currentUserId: string, allocateRoleData: AllocateRoleData): any {
+export function toDenySARoleAssignmentBody(
+  currentUserId: string, allocateRoleData: AllocateRoleData, extraAttributesForBasicRole: {[x: string]: string | boolean} = {}
+): any {
```
Comment (aamirhm) [inline api/roleAccess/dtos/to-role-assignment-dto.ts:97]: Do we allow 'any' datatype?

File: src/work-allocation/containers/my-access/my-access.component.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export class MyAccessComponent extends WorkCaseListWrapperComponent {
           isNew: false
         }).subscribe(() => item.isNew = false);
       } else if (item.role.startsWith('specific-access')) {
-        CasesService.updateChallengedAccessRequestAttributes(this.httpClient, item.case_id, {
+        CasesService.updateSpecificAccessRequestAttributes(this.httpClient, item.case_id, {
           isNew: false
         }).subscribe(() => item.isNew = false);
```
Comment (aamirhm) [inline src/work-allocation/containers/my-access/my-access.component.ts:59]: Please unsubscribe from the subscription to avoid a memory leak.

**Review by aamirhm — CHANGES_REQUESTED** [review]

**Review by udaydenduluri33 — APPROVED** [review]
LGTM

File: src/work-allocation/containers/my-access/my-access.component.ts (hunk 2 above)
Comment (MateiHMCTS) [reply src/work-allocation/containers/my-access/my-access.component.ts]: Agree, needs take(1)

File: api/roleAccess/dtos/to-role-assignment-dto.ts (hunk 1 above)
Comment (MateiHMCTS) [reply api/roleAccess/dtos/to-role-assignment-dto.ts]: No, we should never use any.
it was there before, so I did not give it attention.
I have modified all the file to not have any "any" type

**Review by aamirhm — APPROVED** [review]

**Pull request merged**